From 8f8dc4a86b350043f0e46c2c0b72acb2b5772c38 Mon Sep 17 00:00:00 2001
From: Naga Sureshkumar Relli <naga.sureshkumar.relli@xilinx.com>
Date: Tue, 25 Feb 2020 23:21:44 -0700
Subject: [PATCH 1241/1852] Revert "mtd: rawnand: nand_micron: Do not over
 write driver's (read/write)_page() hooks"

commit c81db8787d8a7b095114d1313abc7afd02d53408 from
https://github.com/Xilinx/linux-xlnx.git

This patch reverts commit a70a639347ac ("mtd: rawnand: nand_micron: Do
not over write driver's (read/write)_page() hooks"). in
pl353_nand_exec_op_cmd() we are not properly calculating the column and
row address of the flash. Hence revert this patch and fix it in the driver

Signed-off-by: Naga Sureshkumar Relli <naga.sureshkumar.relli@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/mtd/nand/raw/nand_micron.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/mtd/nand/raw/nand_micron.c b/drivers/mtd/nand/raw/nand_micron.c
index 541fa293c486..56654030ec7f 100644
--- a/drivers/mtd/nand/raw/nand_micron.c
+++ b/drivers/mtd/nand/raw/nand_micron.c
@@ -501,10 +501,8 @@ static int micron_nand_init(struct nand_chip *chip)
 		chip->ecc.size = 512;
 		chip->ecc.strength = chip->base.eccreq.strength;
 		chip->ecc.algo = NAND_ECC_BCH;
-		if (!chip->ecc.read_page)
-			chip->ecc.read_page = micron_nand_read_page_on_die_ecc;
-		if (!chip->ecc.write_page)
-			chip->ecc.write_page = micron_nand_write_page_on_die_ecc;
+		chip->ecc.read_page = micron_nand_read_page_on_die_ecc;
+		chip->ecc.write_page = micron_nand_write_page_on_die_ecc;
 
 		if (ondie == MICRON_ON_DIE_MANDATORY) {
 			chip->ecc.read_page_raw = nand_read_page_raw_notsupp;
-- 
2.31.1

